src/utils/codeUtils.coffee

```diff
@@ -20,3 +20,26 @@ exports.minLocation = (locations) ->
     locations.forEach (loc) ->
         if exports.compareLocations(loc, min) < 0 then min = loc
     return min
+
+# Given a coffee-script source string, returns the string with all single line comments representing
+# pragmas replaced into herecomments, so that they will be preserved when building the AST.
+exports.pragmaToHereComment = (source) ->
+    if !source or source.length is 0 then return ''
+
+    result = []
+    re = /^(\s*)#\s*(!pragma.+|istanbul\s+ignore.+)\s*$/
```

Comment:
The only weird corner case here would be something like:

    """
    # !pragma ...
    """

Which, I admit, is unlikely...